api/stemcell_service.go

```diff
@@ -50,14 +49,5 @@ func (us UploadStemcellService) Upload(input StemcellUploadInput) (StemcellUploa
 
 	us.progress.End()
 
-	if resp.StatusCode != http.StatusOK {
-		out, err := httputil.DumpResponse(resp, true)
-		if err != nil {
-			return StemcellUploadOutput{}, fmt.Errorf("request failed: unexpected response: %s", err)
-		}
-
-		return StemcellUploadOutput{}, fmt.Errorf("request failed: unexpected response:\n%s", out)
-	}
-
-	return StemcellUploadOutput{}, nil
+	return StemcellUploadOutput{}, ValidateStatusOK(resp)
```

Comment:
@zachgersh Got it. 

Do you also prefer `err` assignment outside of the `if` conditional clause as in the example above? 

I have inlined the `err` assignment in other places where `ValidateStatusOk` was not in return (Eg. [here](https://github.com/pivotal-cf/om/pull/30/commits/d9f2ac2e916016353df8058afacc50635aab64b8#diff-3838a81da13a8c04458b725edd0e4045R93)). I can move them into different lines if that is the style preference.